eng/pipelines/templates/jobs/mgmt-release.yml

```diff
@@ -25,52 +28,55 @@ jobs:
         artifact: packages
         displayName: Download published Nuget Packages
 
-      - task: Powershell@2
-        displayName: 'Verify Package Tags and Create Git Releases'
-        inputs:
-          filePath: ${{ parameters.BuildToolsPath }}/scripts/create-tags-and-git-release.ps1
-          arguments: >
-            -artifactLocation '$(PIPELINE.WORKSPACE)/packages'
-            -workingDirectory $(System.DefaultWorkingDirectory)
-            -packageRepository 'Nuget'
-            -releaseSha '$(Build.SourceVersion)'
-            -repoOwner 'Azure'
-            -repoName 'azure-sdk-for-net'
-          pwsh: true
-        env:
-          GH_TOKEN: $(azuresdk-github-pat)
-        condition: and(succeeded(), ne(variables['SkipCreateTagAndGitRelease'], 'true'))
+      - ${{ if and(eq(parameters.ShouldPublish, 'true'), eq(variables['Build.Repository.Name'], 'Azure/azure-sdk-for-net')) }}:
+        - task: Powershell@2
+          displayName: 'Verify Package Tags and Create Git Releases'
+          inputs:
+            filePath: ${{ parameters.BuildToolsPath }}/scripts/create-tags-and-git-release.ps1
+            arguments: >
+              -artifactLocation '$(PIPELINE.WORKSPACE)/packages'
+              -workingDirectory $(System.DefaultWorkingDirectory)
+              -packageRepository 'Nuget'
+              -releaseSha '$(Build.SourceVersion)'
+              -repoOwner 'Azure'
+              -repoName 'azure-sdk-for-net'
+            pwsh: true
+          env:
+            GH_TOKEN: $(azuresdk-github-pat)
+          condition: and(succeeded(), ne(variables['SkipCreateTagAndGitRelease'], 'true'))
 
       - template: pipelines/steps/net-signing.yml@azure-sdk-build-tools
```

Comment:
Let's move this signing steps to be the first thing in this job and then we can condition all the other steps (except artifact publishing) with the condition you are using. That way we don't have to duplicate that condition so much.

Reply:
Looks like you correctly grouped the bottom set of steps but I still think we can group the tagging with the below steps by moving the signing to be the first step.

Reply:
Sorry I failed to push the latest changes.